### PR TITLE
CRDs and godoc reference for main projects

### PR DESCRIPTION
**Pull request by SafeEHA**
- Created a sidebar structure in SUMMARY.md for main project reference guides
- Added https://pkg.go.dev/ and https://doc.crds.dev/ links to main projects

cc:@lentzi90

### CODE REVIEW

File: docs/user-guide/src/reference/bmo.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Bare Metal Operator Reference
+
+- Baremetal Operator (CRDs): [documentation](https://doc.crds.dev/github.com/metal3-io/baremetal-operator@v0.5.0)
```
Comment (lentzi90) [inline docs/user-guide/src/reference/bmo.md:3]: We have a newer patch release that we could link to

```suggestion
- Baremetal Operator (CRDs): [documentation](https://doc.crds.dev/github.com/metal3-io/baremetal-operator@v0.5.1)
```

**Review by lentzi90 — COMMENTED** [review]
Looks great!
Can you take a look at the markdownlint error? I think there are some trailing newlines that makes it upset :slightly_smiling_face:

**Review by lentzi90 — COMMENTED** [review]
Great!
Please squash the commits, then we can merge
/approve

**Comment by tuminoid** [comment]
> Great! Please squash the commits, then we can merge /approve

Can you squash the commits, so we can merget this? @SafeEHA

**Comment by SafeEHA** [comment]
Done @tuminoid apologies this took so long

**Review by tuminoid — COMMENTED** [review]
/lgtm

Thanks for the contribution!

File: docs/user-guide/src/SUMMARY.md [hunk 2]
```diff
@@ -36,3 +36,7 @@
    - [Try it](developer_environment/tryit.md)
 - [Version Support](version_support.md)
 - [Project Security Policy](security_policy.md)
+- [Refernce]()
```
Comment (dtantsur) [inline docs/user-guide/src/SUMMARY.md:39]: Typo

File: docs/user-guide/src/SUMMARY.md [hunk 3]
```diff
@@ -36,3 +36,7 @@
...
+   - [Baremetal Operator Reference](reference/bmo.md)
+   - [Cluster API provider Refernce](reference/capm3.md)
```
Comment (dtantsur) [inline docs/user-guide/src/SUMMARY.md:41]: Typo

File: docs/user-guide/src/reference/bmo.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# Bare Metal Operator Reference
```
Comment (dtantsur) [inline docs/user-guide/src/reference/bmo.md:1]: Why create new pages for 2 lines of information? Let's maybe consolidate all of these under the parent "References" document?

**Comment by dtantsur** [comment]
Late to the party, but could someone please address the feedback?

File: docs/user-guide/src/reference/bmo.md (hunk 4 above)
Comment (SafeEHA) [reply docs/user-guide/src/reference/bmo.md:1]: I assumed there would be more in the future but if that's not the case are you asking that only a Reference be created and all 3 resources be under that one page?

**Pull request merged**